DR-11 Add specification README. (#691)
blindnet-io/product-management: specifications/README.md

# Specification

> The first step of building products at blindnet always includes specification writing. It helps not only us to faster build better products, but we hope it also helps you to better and faster understand how our products work.

Reviewer: ```suggestion
The first step of building products at blindnet always includes specification writing. It helps not only us to rapidly build better products, but we hope it also helps others better understand how our products work.
```

- Activity diagrams, defining the product through user actions and activities.
- State machine diagrams, defining how the product behaves in relation to different user inputs.
- User interface sketches, defining the look of user-facing parts of the product.

## Technical specification

Technical specification defines the product from the implementational point of view. The main parts of technical specification are functional requirements, architecture, and design.

### Functional requirements

Functional requirements define functional aspects of the product and are captured in the functional requirements document (FRD). Each functional requirement defines one particular function a product needs to deliver, and is assigned a unique code identifier (e.g., FRD-BE01, FR-SDK14). 

### Architecture

Product architecture defines different components of the product structure and how they interact together in order to deliver specified functions. For example, the architecture document defines different SDKs, APIs and databases that, when put together, deliver promised functional aspects. 

### Design

> Product design defines in detail the technical aspects of each architectural component of the product. It includes, for example, database models, or expected inputs/outputs of SDK functions and REST API routes. In some cases, like for example for REST API routes, the design is described together with the corresponding functional requirement. 

Reviewer: ```suggestion
Product design defines in detail the technical aspects of each architectural component of the product. It includes, for example, database models, or expected inputs/outputs of SDK functions and REST API endpoints. In some cases, like for example for REST API endpoints, the design is described together with the corresponding functional requirement. 
```
